packages/web3torrent/src/utils/checkTorrentInTracker.ts

```diff
@@ -1,33 +1,29 @@
 import {Client} from 'bittorrent-tracker';
 import {defaultTrackerOpts} from '../constants';
 import {TorrentTestResult} from '../library/types';
-import debug from 'debug';
-const log = debug('web3torrent:tracker-check');
+import {logger} from '../logger';
+const log = logger.child({module: 'tracker-check'});
 
 export async function checkTorrentInTracker(infoHash: string) {
-  log(`Scraping tracker for torrent ${infoHash}`);
-  const trackerClient: Client = new Client({
-    ...defaultTrackerOpts,
-    infoHash: [infoHash]
-  });
+  log.info(`Scraping tracker for torrent ${infoHash}`);
+
+  const client = new Client({...defaultTrackerOpts, infoHash: [infoHash]});
   const trackerScrape: Promise<number> = new Promise(resolve => {
-    trackerClient.once('scrape', data => resolve(data.complete));
-    setTimeout(() => trackerClient.scrape(), 2500); // waits ~2 seconds as that's the tracker update tick
+    client.once('scrape', data => resolve(data.complete));
+    setTimeout(() => client.scrape(), 2500); // waits ~2 seconds as that's the tracker update tick
+    setTimeout(() => resolve(-1), 5000); // returns -1 if there was no answer from the tracker(s) for ~2.5 seconds
```

Comment:
should we use clearTimeout in case this callback it loses the race?

Reply:
Are you talking about the first setTimeout or the second?
For the second, I don't think it's necessary, there's no side-effect there, and once a promise resolves, all subsequent tries to resolve simply don't have any effect.

If you are talking about the first one (`setTimeout(() => client.scrape(), 2500);`) I think it's impossible for the other setTimeout to run before this one. And the possible side-effects are handled after with the `client.stop()` and `client.destroy()`

Reply:
I think your reasoning is solid. Good thinking!